### PR TITLE
Install detekt

### PR DESCRIPTION
**Pull request by matsumo0922**
## Issue
- #177 

## Overview (Required)
- Install detekt
- ~Uninstall spotless linter~

## Required review point
- ~Is it ok to uninstall the spotless linter~
- I have set up a review using Review Dog in Format Job in Github Actions. However, if there is a lot of reviews, the PR may become difficult to read.
- Please give me a review of Detekt's rules...

## FYI
- This PR does not run detekt because a lot of differences will be generated. I will fix it in the next PR.

### CODE REVIEW

File: .github/workflows/Format.yml
Comment (takahirom) [inline .github/workflows/Format.yml:27]: This PR will remove the KtLint, so we can't autoformat anymore, right?

Comment (takahirom) [reply .github/workflows/Format.yml:27]: There is a "autoCorrect". does it work? 👀

Comment (matsumo0922) [reply .github/workflows/Format.yml:27]: Yes, it's work.
The `detekt-formatting` library is a wrapper for ktlint, so we can continue to use auto format.

https://detekt.dev/docs/intro

**Comment by takahirom** [comment]
> This PR does not run detekt because a lot of differences will be generated. I will fix it in the next PR.

Could you elaborate the plan?
Do we have to merge this first and wait the next PR?

**Comment by matsumo0922** [comment]
I was thinking of running detekt on a different PR since it would be difficult to see the differences if I run detekt on this PR.

As soon as this PR is merged, we need to quickly create a PR that runs detekt and merge it. (to the consternation of other developers).

---

このPRでdetektを実行してしまうと、差分が見づらくなりレビューがしづらくなると思ったため別のPRで実行しようと考えていました。

このPRをマージすると unformat なコードが master にマージされてしまうため、マージされ次第すぐに detekt を実行したPRを作り、マージする必要があります。

もしくは feature ブランチに一度マージし、detekt を実行した PR もそこにマージしてから master に入れるのがいいのかもしれません（今見たら宛先が変えられなくなっているので無理かもしれませんが）

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/AndroidGradleDsl.kt
Comment (takahirom) [inline build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/AndroidGradleDsl.kt:73]: We are using English for comments.
Could you use English?🙇
You can use DeepL or something.

**Comment by takahirom** [comment]
How about leaving the CI setting of spotless? I think we can merge this if we can run format.👀

File: build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/AndroidGradleDsl.kt
Comment (matsumo0922) [reply build-logic/src/main/kotlin/io/github/droidkaigi/confsched2023/primitive/AndroidGradleDsl.kt]: Fixed. 67fd868

**Comment by matsumo0922** [comment]
Indeed, it looks good.
Reverted. f38b304

File: .github/workflows/WorkflowHook.yml
Comment (takahirom) [inline .github/workflows/WorkflowHook.yml:103]: Maybe we have to use spotlessKotlinApply for now? 👀

Comment (matsumo0922) [reply .github/workflows/WorkflowHook.yml]: Fixed~ e0a4772

**Comment by matsumo0922** [comment]
@tomoya0x00 
I would like a you're review. :bow:

**Review by takahirom — APPROVED** [review]
Thanks for your challenge! I'm looking forward to seeing your next PR!

**Pull request merged**
